packages/bits/src/services/unit-conversion.service.ts

```diff
@@ -72,10 +78,21 @@ export class UnitConversionService {
     public getFullDisplay(conversion: IUnitConversionResult, unit: UnitOption, plusSign = false, nanDisplay = "---"): string {
         const isValidNumber = this.isValidNumber(conversion.value);
         const spacing = unit !== "generic" && isValidNumber ? " " : "";
-        const unitDisplay = isValidNumber ? this.getUnitDisplay(conversion, unit) : "";
+        let unitDisplay = isValidNumber ? this.getUnitDisplay(conversion, unit) : "";
 
         // The generic unit is not currently i18n friendly
         const localizeValue = unit !== "generic";
+
+        if (!unitDisplay && !(conversion.value === "NaN") && conversion.order ) {
+            const tempObj = {
+                ...conversion,
+            };
+            tempObj.order = 0;
+            unitDisplay = this.getUnitDisplay(tempObj, unit)
+            const prefix = plusSign && parseInt(conversion.value, 10) > 0 ? "+" : "";
+
+            return `${prefix}${conversion.scientificNotation}${spacing}${unitDisplay}`;
+        }
         return `${this.getValueDisplay(conversion, plusSign, nanDisplay, localizeValue)}${spacing}${unitDisplay}`;
```

Comment:
What do you think about creating a getScientificDisplay method that encapsulates the prefix handling like getValueDisplay does? I did something like that below with a little additional cleanup:

```suggestion
        
        let displayValue = this.getValueDisplay(conversion, plusSign, nanDisplay, localizeValue);
        if (!unitDisplay && conversion.value !== "NaN" && conversion.order) {
            const conversionCopy = { ...conversion };
            conversionCopy.order = 0;
            unitDisplay = this.getUnitDisplay(tempObj, unit)
            displayValue = this.getScientificDisplay(conversion, plusSign, nanDisplay, localizeValue);
        }
        return `${displayValue}${spacing}${unitDisplay}`;
```